src/vaadin-date-picker-mixin.html

```diff
@@ -344,7 +341,7 @@
     disconnectedCallback() {
       super.disconnectedCallback();
 
-      if (this.$.overlay) {
+      if (!this.$.overlay.hasAttribute('disable-upgrade')) {
```

Comment:
consider adding a property, e.g. `_overlayInitialized` and using it instead

Reply:
Updated.